imagebuildah/stage_executor.go

```diff
@@ -1850,15 +1875,66 @@ func (s *StageExecutor) getCreatedBy(node *parser.Node, addedContentSummary stri
 			}
 		}
 		buildArgs := s.getBuildArgsKey()
-		return "/bin/sh -c #(nop) ARG " + buildArgs
+		return "/bin/sh -c #(nop) ARG " + buildArgs, nil
 	case "RUN":
 		shArg := ""
 		buildArgs := s.getBuildArgsResolvedForRun()
+		mountOptionSource := ""
+		mountOptionFrom := ""
+		appendCheckSum := ""
+		var err error
+		for _, flag := range node.Flags {
+			if strings.HasPrefix(flag, "--mount") {
+				mountOptionSource, mountOptionFrom = internalParse.GetFromAndSourceKeysFromMountFlag(flag)
+				// If source is not specified then default is '.'
+				if mountOptionSource == "" {
+					mountOptionSource = "."
+				}
+			}
+		}
+		// Source specificed is part of stage, image or additional-build-context.
+		if mountOptionFrom != "" {
+			// If this is not a stage then get digest of image or additional build context
+			if _, ok := s.executor.stages[mountOptionFrom]; !ok {
+				if builder, ok := s.executor.containerMap[mountOptionFrom]; ok {
+					// Found valid image, get image digest.
+					appendCheckSum = builder.FromImageDigest
+				} else {
+					if s.executor.additionalBuildContexts[mountOptionFrom].IsImage {
+						if builder, ok := s.executor.containerMap[s.executor.additionalBuildContexts[mountOptionFrom].Value]; ok {
+							// Found valid image, get image digest.
+							appendCheckSum = builder.FromImageDigest
+						}
+					} else {
+						// Found additional build context, get directory sha.
+						basePath := s.executor.additionalBuildContexts[mountOptionFrom].Value
+						if s.executor.additionalBuildContexts[mountOptionFrom].IsURL {
+							basePath = s.executor.additionalBuildContexts[mountOptionFrom].DownloadedCache
+						}
+						appendCheckSum, err = internalUtil.GeneratePathChecksum(filepath.Join(basePath, mountOptionSource))
+						if err != nil {
+							return "", fmt.Errorf("Unable to generate checksum for image history: %w", err)
```

Comment:
Text passed to fmt.Errorf() is context, not a full error message, so it should not be capitalized, and should merely describe what was being attempted, so that the caller can add more context if it wishes.  "generating checksum for directory %q in %q: %w" would be enough here, since the caller will prepend context of its own.  Not just here, but throughout this PR.